nemo/collections/asr/parts/submodules/multi_head_attention.py

```diff
@@ -156,15 +147,11 @@ def forward(self, query, key, value, mask, pos_emb=None, cache=None, cache_next=
 
     def update_cache(self, key, value, query, cache, cache_next):
         if cache is not None:
-            q_length = query.size(1)
-            q_input = query
-            key = value = torch.cat((cache[self._cache_id], key), dim=1)
-
-        if cache_next is not None:
-            q_keep_size = torch.tensor(q_length - self.cache_drop_size, dtype=torch.int64).clip(min=1)
-            keep_in_cache_next(cache=cache, cache_next=cache_next, keep_size=q_keep_size, cache_id=self._cache_id)
-            cache_next[self._cache_id, :, -q_keep_size:, :] = q_input[:, :q_keep_size, :]
-
+            key = value = torch.cat([cache[self._cache_id], key], dim=1)
```

Comment:
Let's add "cache is not None and cache_next is not None" to the if block or separate cache and cache_next like before. Both ways are OK.